### PR TITLE
Use `ArgumentException.ThrowIfNullOrEmpty` part 1

### PR DESCRIPTION
**Pull request by xtqqczze**
Use `ArgumentException.ThrowIfNullOrEmpty` in _Microsoft.PowerShell.Commands.Management_.

Contributes to #19212.

### Changes

* Throws `ArgumentException` instead of `ArgumentNullException` when a string is empty.
  * Possible breaking change as `ArgumentException` is a less derived type.
* Throws `ArgumentNullException` instead of `ArgumentException` when a string is null.
  * Clearly non-breaking as `ArgumentNullException` is a more derived type.

### Breaking changes contract

_[Bucket 4: Clearly Non-Public](https://github.com/PowerShell/PowerShell/blob/master/docs/dev-process/breaking-change-contract.md#bucket-4-clearly-non-public)_

* Modifies internal class `Microsoft.PowerShell.Cmdletization.Cim.CimQuery`
* Modifies internal class `Microsoft.PowerShell.Commands.CIMHelper`

### Other changes

608186d0a3524a986f471f2db60058d5f63c7f10 is a related refactoring that does not change behavior.

### CODE REVIEW

**Comment by daxian-dbw** [comment]
@xtqqczze The change is for adopting `ArgumentException.ThrowIfNullOrEmpty` instead of `ArgumentNullException.ThrowIfNull`, right?

**Comment by xtqqczze** [comment]
> @xtqqczze The change is for adopting `ArgumentException.ThrowIfNullOrEmpty` instead of `ArgumentNullException.ThrowIfNull`, right?

My bad, this is why I'm trying to keep PRs small.

**Comment by xtqqczze** [comment]
> > @xtqqczze The change is for adopting `ArgumentException.ThrowIfNullOrEmpty` instead of `ArgumentNullException.ThrowIfNull`, right?
> 
> My bad, this is why I'm trying to keep PRs small.


Although I've mixed the following changes when they could have been seperated.:
* Throws `ArgumentException` instead of `ArgumentNullException` when a string is empty.
* Throws `ArgumentNullException` instead of `ArgumentException` when a string is null.

In any case the changes are to non-public APIS.

File: src/Microsoft.PowerShell.Commands.Management/cimSupport/cmdletization/cim/cimQuery.cs [hunk 1]
```diff
@@ -314,11 +314,8 @@ public override void FilterByAssociatedInstance(object associatedInstance, strin
         /// <param name="optionValue"></param>
         public override void AddQueryOption(string optionName, object optionValue)
         {
-            if (string.IsNullOrEmpty(optionName))
-            {
-                throw new ArgumentNullException(nameof(optionName));
-            }
-
+            ArgumentException.ThrowIfNullOrEmpty(optionName);
+
```
Comment (CarloToso) [inline src/Microsoft.PowerShell.Commands.Management/cimSupport/cmdletization/cim/cimQuery.cs:318]:
```suggestion
```

Comment (xtqqczze) [reply src/Microsoft.PowerShell.Commands.Management/cimSupport/cmdletization/cim/cimQuery.cs:318]: My personal preference is to separate the method calls with a blank line when the type differs.

Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Management/cimSupport/cmdletization/cim/cimQuery.cs:318]: I like it, I'll update my PR following your style 👍

**Review by iSazonov — APPROVED** [review]

File: src/Microsoft.PowerShell.Commands.Management/cimSupport/cmdletization/cim/cimQuery.cs (hunk 1 above)
Comment (daxian-dbw) [reply src/Microsoft.PowerShell.Commands.Management/cimSupport/cmdletization/cim/cimQuery.cs:318]: I think it's better to group them together as they are doing the same kind of things. For example, we group `Dbg.Assert(xxx)` together at the beginning of some methods, even though they do different checks.

**Review by daxian-dbw — APPROVED** [review]
LGTM

**Pull request merged**